boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,201 @@
+
+// The SPI handle for the SPI device the E-Fuses are connected to
+static SPI_HandleTypeDef *efuse_spi_handle;
+
+void Io_Efuse_Init(SPI_HandleTypeDef *const hspi)
+{
+    efuse_spi_handle = hspi;
+}
+
+void Io_Efuse_ConfigureEfuse(struct Efuse *e_fuse)
+{
+    // Global config register
+    Io_Efuse_WriteReg(SI_GCR_ADDR, GCR_CONFIG, e_fuse);
+
+    // Channel 0 config registers
+    Io_Efuse_WriteReg(SI_RETRY_0_ADDR, RETRY_0_CONFIG, e_fuse);
+    Io_Efuse_WriteReg(SI_CONFR_0_ADDR, CONFR_0_CONFIG, e_fuse);
+    Io_Efuse_WriteReg(SI_OCR_0_ADDR, OCR_0_CONFIG, e_fuse);
+
+    // Channel 1 config registers
+    Io_Efuse_WriteReg(SI_RETRY_1_ADDR, RETRY_1_CONFIG, e_fuse);
+    Io_Efuse_WriteReg(SI_CONFR_1_ADDR, CONFR_1_CONFIG, e_fuse);
+    Io_Efuse_WriteReg(SI_OCR_1_ADDR, OCR_1_CONFIG, e_fuse);
+}
+
+void Io_Efuse_ConfigureChannelMonitoring(
+    uint8_t       selection,
+    struct Efuse *e_fuse)
+{
+    uint16_t reg_val = 0x0000;
+
+    // Read original content of GCR Register
+    reg_val = Io_Efuse_ReadReg(SI_GCR_ADDR, e_fuse);
+
+    CLEAR_BIT(reg_val, (CSNS1_EN_MASK | CSNS0_EN_MASK));
+    SET_BIT(reg_val, (selection & (CSNS1_EN_MASK | CSNS0_EN_MASK)));
+
+    Io_Efuse_WriteReg(SI_GCR_ADDR, reg_val, e_fuse);
+}
+
+void Io_Efuse_UpdateStatus(struct Efuse *e_fuse)
+{
+    uint16_t status = StatusType_NoFault;
+
+    status = Io_Efuse_ReadReg(SO_STATR_ADDR, e_fuse);
+
+    e_fuse->status = (StatusType_e)status;
+}
+
+void Io_Efuse_UpdateFaults(struct Efuse *e_fuse)
+{
+    uint16_t channel_0_faults = FaultType_NoFault;
+    uint16_t channel_1_faults = FaultType_NoFault;
+
+    channel_0_faults = Io_Efuse_ReadReg(SO_FAULTR_0_ADDR, e_fuse);
+    channel_1_faults = Io_Efuse_ReadReg(SO_FAULTR_1_ADDR, e_fuse);
+
+    e_fuse->faults.channel_0_faults = (FaultType_e)channel_0_faults;
+    e_fuse->faults.channel_1_faults = (FaultType_e)channel_1_faults;
+}
+
+void Io_Efuse_WriteReg(
+    uint8_t       register_address,
+    uint16_t      register_value,
+    struct Efuse *e_fuse)
+{
+    HAL_StatusTypeDef status;
+    uint16_t          command        = 0x0000U;
+
+    // Place the register address into bits 10->13
+    command =
+        (uint16_t)((register_address & EFUSE_ADDR_MASK) << EFUSE_ADDR_SHIFT);
+    // Place register value to be written into bits 0->8
+    command = (uint16_t)(command | (register_value & EFUSE_SI_DATA_MASK));
+    // Invert watchdog bit state
+    if (e_fuse->watch_dog_state)
+    {
+        SET_BIT(command, WATCH_DOG_BIT);
+    }
+    else
+    {
+        CLEAR_BIT(command, WATCH_DOG_BIT);
+    }
+    // Invert watchdog bit state for next write
+    e_fuse->watch_dog_state = !e_fuse->watch_dog_state;
+
+    // Compute and set/clear parity value
+    Io_Efuse_SetParityBit(&command);
+
+        status = Io_Efuse_WriteToEfuse(
+            &command, e_fuse->chip_select.GPIO_Port,
+            e_fuse->chip_select.GPIO_Pin);
+    UNUSED(status);
+}
+
+uint16_t Io_Efuse_ReadReg(uint8_t register_address, struct Efuse *e_fuse)
```

Comment:
```suggestion
HAL_StatusTypeDef Io_Efuse_ReadReg(uint8_t register_address, struct Efuse *e_fuse, uint16_t* returned_register)
```
This way we can return status

Reply:
Haha ok it used to be called _Io_Efuse_ReadRegister()_ but you made the suggestion to change it to _Io_Efuse_ReadReg()_